src/libxtp/dftengine/dftengine.cc

```diff
@@ -1061,5 +1070,19 @@ namespace votca {
 
             return avdmat;
         }
+
+        
+        
+        void DFTENGINE::CalculateERIs_4c(const AOBasis& dftbasis, const ub::matrix<double> &DMAT) {
+
+          if (_with_RI)
+            _ERIs.CalculateERIs(_dftAOdmat);
```

Comment:
I like that you use more functions and make them smaller, I would prefer if the input is checked in initialize and not later, e.g. there is a method in tools : ` T ifExistsAndinListReturnElseThrowRuntimeError(const string &key,std::vector<T> possibleReturns); `, 

which can be called  e.g.
`std::vector<std::string> choices={"direct","cache","RI"};
`result=ifExistsAndinListReturnElseThrowRuntimeError(input ,choices); `